src/core/catalog.js

```diff
@@ -1285,7 +1289,14 @@ class Catalog {
       // node further down in the tree (see issue5644.pdf, issue8088.pdf),
       // and to ensure that we actually find the correct `Page` dict.
       for (let last = kids.length - 1; last >= 0; last--) {
-        nodesToVisit.push(kids[last]);
+        const lastKid = kids[last];
+        nodesToVisit.push(lastKid);
+
+        // Launch all requests in parallel so we don't wait for each one in turn
+        // when looking for a page near the end, if all the pages are top level
```

Comment:
Nit: Period at the end of sentences please.
```suggestion
        // when looking for a page near the end, if all the pages are top level.
```